rocAL/rocAL/source/augmentations/geometry_augmentations/node_crop.cpp

```diff
@@ -27,68 +27,79 @@ THE SOFTWARE.
 #include "exception.h"
 
 CropNode::CropNode(const std::vector<Tensor *> &inputs, const std::vector<Tensor *> &outputs) :
-        Node(inputs, outputs),
-        _dest_width(_outputs[0]->info().max_shape()[0]),
-        _dest_height(_outputs[0]->info().max_shape()[1])
-{
+        Node(inputs, outputs) {
     _crop_param = std::make_shared<RocalCropParam>(_batch_size);
 }
 
-void CropNode::create_node()
-{
+void CropNode::create_node() {
     if(_node)
         return;
 
-    if(_dest_width == 0 || _dest_height == 0)
-        THROW("Uninitialized destination dimension")
-
     _crop_param->create_array(_graph);
+    create_crop_tensor(_crop_tensor, &_crop_coordinates);
```

Comment:
I am not sure why we need to create_crop_tensor here. I don't see it  is getting updated. Also we can't create crop tensor in node.h base class.

Reply:
We are creating a crop tensor from handle(i.e _crop_coordinates)
In the [update node](https://github.com/fiona-gladwin/MIVisionX/blob/ac0bed6de0df4bf2848ad5e3d333579c2fa00ffc/rocAL/rocAL/source/augmentations/geometry_augmentations/node_crop.cpp#L57) we are updating the values in the handle which will be reflected in the tensor as well.

The crop tensor is present only in the derived nodes only the API to create in present in the node.h as it can be used by other crop related nodes.

Reply:
As discussed, Removed this function from node class.

- Introduced the create_crop_tensor only in CropNode
- For all crop related augmentations inheriting the class from CropNode
- Added the crop_tensor and crop_coordinates as protected variables so it can be accessed by all the derived class.
- The destructor is added only for crop nodes and removed from other crop related augmentations